app/components/auth/signUp/components/surveyForm/components/question.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { findIndex, unionBy } from 'lodash';
+import { RawQuestionType, Survey, SCINAPSE_SURVEY_NAME } from '../constants';
+import { withStyles } from '../../../../../../helpers/withStylesHelper';
+const styles = require('./question.scss');
+
+interface QuestionProps {
+  question: RawQuestionType;
+  qKey: number;
+  surveyResult: Survey;
+  handleSetSurveyResult: (value: React.SetStateAction<Survey>) => void;
+}
+
+interface AnswerProps {
+  value: string;
+  name: string;
+  type: string;
+  handleCheckChange: () => void;
+}
+
+function changeSurveyAnswer(
+  surveyPayload: Survey,
+  type: string,
+  surveyResult: Survey,
+  handleSetSurveyResult: (value: React.SetStateAction<Survey>) => void
+) {
+  if (!surveyResult) {
+    return handleSetSurveyResult(surveyPayload);
+  }
+
+  const questionPayload = surveyPayload.questions[0];
+  const targetSurveyIndex = findIndex(surveyResult.questions, ['question', questionPayload.question]);
+
+  if (targetSurveyIndex >= 0) {
+    const targetSurvey = surveyResult.questions[targetSurveyIndex];
+
+    if (type === 'checkbox') {
+      const targetAnswerIndex = findIndex(targetSurvey.checked, questionPayload.checked[0]);
```

Comment:
this is too verbose and hard to understand. and also, doesn't be needed to solve this problem.
furthermore, this function couldn't handle anything other than a checkbox.

I suggest simply make a `selected` or sth like that state, and just change the pointer.